### PR TITLE
added symbol check when checking for identifier

### LINKED ISSUE #727: Type from registered type has higher preference than input parameter expression with same name
## 1. Description
When registering a Type with `[DynamicLinqType]` or with adding it in custom type provider, it gains precedence over passed input parameters. Even if the case is different from Type, type get higher preference.

Ideally input parameters should have highest priority.

## 2. Exception

```
Exception message:Unhandled exception. '.' or '(' or string literal expected (at index 3)
Stack trace:   at System.Linq.Dynamic.Core.Tokenizer.TextParser.ValidateToken(TokenId tokenId, String errorMessage) in C:\Dev\GitHub\System.Linq.Dynamic.Core\src\System.Linq.Dynamic.Core\Tokenizer\TextParser.cs:line 451
   at System.Linq.Dynamic.Core.Parser.ExpressionParser.ParseTypeAccess(Type type, Boolean getNext) in C:\Dev\GitHub\System.Linq.Dynamic.Core\src\System.Linq.Dynamic.Core\Parser\ExpressionParser.cs:line 1658
   at System.Linq.Dynamic.Core.Parser.ExpressionParser.ParseIdentifier() in C:\Dev\GitHub\System.Linq.Dynamic.Core\src\System.Linq.Dynamic.Core\Parser\ExpressionParser.cs:line 941
   at System.Linq.Dynamic.Core.Parser.ExpressionParser.ParsePrimaryStart() in C:\Dev\GitHub\System.Linq.Dynamic.Core\src\System.Linq.Dynamic.Core\Parser\ExpressionParser.cs:line 817
   at System.Linq.Dynamic.Core.Parser.ExpressionParser.ParsePrimary() in C:\Dev\GitHub\System.Linq.Dynamic.Core\src\System.Linq.Dynamic.Core\Parser\ExpressionParser.cs:line 785
   at System.Linq.Dynamic.Core.Parser.ExpressionParser.ParseUnary() in C:\Dev\GitHub\System.Linq.Dynamic.Core\src\System.Linq.Dynamic.Core\Parser\ExpressionParser.cs:line 780
   at System.Linq.Dynamic.Core.Parser.ExpressionParser.ParseMultiplicative() in C:\Dev\GitHub\System.Linq.Dynamic.Core\src\System.Linq.Dynamic.Core\Parser\ExpressionParser.cs:line 725
```

## 3. Fiddle or Project
If you are able,

Provide a Fiddle that reproduce the issue: https://dotnetfiddle.net/6nbGZ6

## 4. Any further technical details
Add any relevant detail that can help us.

### PR DESCRIPTION
**Pull request by abbasc52**
Fixes #727

### CODE REVIEW

File: src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs [hunk 1]
```diff
@@ -932,7 +932,14 @@ private Expression ParseIdentifier()
         var isValidKeyWord = _keywordsHelper.TryGetValue(_textParser.CurrentToken.Text, out var value);
 
         var extraCondition = !_parsingConfig.PrioritizePropertyOrFieldOverTheType ||
```
Comment (StefH) [inline src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs:934]: This code gets complex to understand.

Change it to:
```
var isValueAType = value is Type;

var hasPropertyOrField = _it != null && FindPropertyOrField(_it.Type, _textParser.CurrentToken.Text, false) != null;

var hasSymbol = _symbols.ContainsKey(_textParser.CurrentToken.Text);

var shouldNotPrioritizePropertyOrFieldOverTheType = !_parsingConfig.PrioritizePropertyOrFieldOverTheType;
var shouldPrioritizeAndNoMatch = _parsingConfig.PrioritizePropertyOrFieldOverTheType && !(isValueAType && (hasPropertyOrField || hasSymbol));

var extraCondition = shouldNotPrioritizePropertyOrFieldOverTheType || shouldPrioritizeAndNoMatch;
```

**Review by StefH — CHANGES_REQUESTED** [review]

**Comment by StefH** [comment]
@abbasc52 
Can you take a look at my comment?

**Comment by abbasc52** [comment]
Hi @StefH , sure i was busy last week, will try adding it tomorrow

File: src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs (hunk 1 above)
Comment (abbasc52) [reply src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs]: @StefH updated the PR with your suggestion with slight changes to ensure performance is not impacted. Please have a look

**Comment by abbasc52** [comment]
@StefH Can you check if anything else is required?

**Review by StefH — APPROVED** [review]

**Pull request merged**